frontend/js/schema-conversion.js

```diff
@@ -0,0 +1,508 @@
+        }
+        selectHTML += `<select onchange='dataTypeUpdate(id, ${JSON.stringify(globalDataTypeList)})' class='form-control tableSelect' id=${selectId} style='border: 0px !important;'>`
+        for (var k = 0; k < optionsLength; k++) {
+          selectHTML += `<option value='${globalDataTypeList[Object.keys(globalDataTypeList)[i]][k].T}'>${globalDataTypeList[Object.keys(globalDataTypeList)[i]][k].T} </option>`;
+        }
+        selectHTML += `</select></div>`;
+        tableContent += selectHTML + `</td>`;
+      }
+    }
+    tableContent += `</tr>`;
+  }
+
+  globalDataTypeTable = `<table class='data-type-table' id='globalDataTypeTable'>
+                            <tbody>
+                              <tr>
+                                <th>Source</th>
+                                <th>Spanner</th>
+                              </tr>
+                              ${tableContent}
+                            </tbody>
+                         </table>`;
+  document.getElementById('globalDataType').innerHTML = globalDataTypeTable;
+  tooltipHandler();
+}
+
+/**
+ * Function to update data types with warnings(if any) in global data type table
+ *
+ * @param {string} id id of select box in global data type table
+ * @return {null}
+ */
+const dataTypeUpdate = (id, globalDataTypeList) => {
+  let idNum = parseInt(id.match(/\d+/), 10);
+  let dataTypeOptionArray = globalDataTypeList[document.getElementById('dataTypeKey' + idNum).innerHTML];
+  let optionHTML = '';
+  let length = dataTypeOptionArray.length;
+  let selectHTML = `<div style='display: flex;'>
+                  <i class="large material-icons warning" style='cursor: pointer; visibility: hidden;'>warning</i>
+                  <select onchange='dataTypeUpdate(id, ${JSON.stringify(globalDataTypeList)})' class='form-control tableSelect' id=${id} style='border: 0px !important;'>
+                </div>`;
+  for (var x = 0; x < length; x++) {
+    let optionFound = dataTypeOptionArray[x].T === document.getElementById(id).value;
+    if (dataTypeOptionArray[x].T === document.getElementById(id).value && dataTypeOptionArray[x].Brief !== "") {
+
+      selectHTML = `<div style='display: flex;'>
+                      <i class="large material-icons warning" style='cursor: pointer;' data-toggle='tooltip' data-placement='bottom' title='${dataTypeOptionArray[x].Brief}'>warning</i>
+                      <select onchange='dataTypeUpdate(id, ${JSON.stringify(globalDataTypeList)})' class='form-control tableSelect' id=${id} style='border: 0px !important;'>
+                    </div>`;
+    }
+    if (optionFound === true) {
+      optionHTML += `<option selected='selected' value='${dataTypeOptionArray[x].T}'>${dataTypeOptionArray[x].T} </option>`;
+    }
+    else {
+      optionHTML += `<option value='${dataTypeOptionArray[x].T}'>${dataTypeOptionArray[x].T} </option>`;
+    }
+  }
+  selectHTML += optionHTML + `</select>`;
```

Comment:
1. selectHTML has extra </div> tag between <select> tag and <options> tag
2. warning tag can be simplified instead of repeating same code check line no 340 to 350

Reply:
1. div tag is removed
2. repeated code removed